src/plugins/usage_collection/README.md

```diff
@@ -1,39 +1,132 @@
 # Kibana Usage Collection Service
 
-Usage Collection allows collecting usage data for other services to consume (telemetry and monitoring).
-To integrate with the telemetry services for usage collection of your feature, there are 2 steps:
+The Usage Collection Service defines a set of APIs for other plugins to report the usage of their features. At the same time, it provides necessary the APIs for other services (i.e.: telemetry, monitoring, ...) to consume that usage data.
 
-1. Create a usage collector.
-2. Register the usage collector.
+## How to report my plugin's usage?
 
-## Creating and Registering Usage Collector
+The way to report the usage of any feature depends on whether the actions to track occur in the UI, or the usage depends on any server-side data. For that reason, the set of APIs exposed in the `public` and `server` contexts are different.
 
-Your usage collector needs to provide 
-- a `type` for organizing your fields, 
-- `schema` field to define the expected types of usage fields reported, 
-- a `fetch` method for returning your usage data, and
-- an `isReady` method (that returns true or false) for letting the telemetry service know if it needs to wait for any asynchronous action (initialization of clients or other services) before calling the `fetch` method. 
+In any case, to use any of these APIs, the plugin must optionally require the plugin `usageCollection`:
 
-Then you need to make the Telemetry service aware of the collector by registering it.
+```json
+// plugin/kibana.json
+{
+  "id": "...",
+  "optionalPlugins": ["usageCollection"]
+}
+```
 
-1. Make sure `usageCollection` is in your optional Plugins:
+Please, be aware that plugins listing `usageCollection` in the `optionalPlugins` list are allowed to run even when `usageCollection` is disabled. However, this also means that it may not be available. Make sure the plugin defines the types of its contract interfaces with `usageCollection` being optional as well.
 
-    ```json
-    // plugin/kibana.json
-    {
-      "id": "...",
-      "optionalPlugins": ["usageCollection"]
-    }
-    ```
+### `public` APIs
+
+The APIs exposed in the `public` context aim to collect the _aggregate_ number of events that occur in a period of time. They are **not** intended for user-behavioural tracking. The APIs available can be categorized in 2: Application Usage and UI Counters.
+
```

Comment:
Maybe we can add to the `public` APIs the UI Settings collector as well? Should we do it now or once we finish up https://github.com/elastic/kibana/issues/85811?